components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/AssetEntityAdapter.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+
+package org.sonatype.nexus.repository.storage;
+
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.sonatype.nexus.blobstore.api.BlobRef;
+import org.sonatype.nexus.common.entity.EntityId;
+import org.sonatype.nexus.orient.OClassNameBuilder;
+
+import com.google.common.collect.ImmutableMap;
+import com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx;
+import com.orientechnologies.orient.core.id.ORID;
+import com.orientechnologies.orient.core.metadata.schema.OClass;
+import com.orientechnologies.orient.core.metadata.schema.OClass.INDEX_TYPE;
+import com.orientechnologies.orient.core.metadata.schema.OType;
+import com.orientechnologies.orient.core.record.impl.ODocument;
+import com.orientechnologies.orient.core.sql.OCommandSQL;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+import static org.sonatype.nexus.repository.storage.StorageFacet.P_BLOB_REF;
+import static org.sonatype.nexus.repository.storage.StorageFacet.P_COMPONENT;
+import static org.sonatype.nexus.repository.storage.StorageFacet.P_CONTENT_TYPE;
+import static org.sonatype.nexus.repository.storage.StorageFacet.P_NAME;
+import static org.sonatype.nexus.repository.storage.StorageFacet.P_SIZE;
+
+/**
+ * {@link Asset} entity-adapter.
+ *
+ * @since 3.0
+ */
+@Named
+@Singleton
+public class AssetEntityAdapter
+    extends MetadataNodeEntityAdapter<Asset>
+{
+  public static final String DB_CLASS = new OClassNameBuilder()
+      .type(Asset.class)
+      .build();
+
+  private static final String I_COMPONENT = DB_CLASS + ".component_idx";
```

Comment:
The docs say "." should be used only for indexes with the format <className>.<property>. It seems to be syntactic sugar for the CREATE INDEX command, doesn't appear to do much from the Java API (since you have to specify class and property anyways).
http://orientdb.com/docs/2.0/orientdb.wiki/SQL-Create-Index.html